src/App.tsx

```diff
@@ -1,26 +1,225 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { deleteTodo, getTodos, updateTodo, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { FilterType } from './types/FilterType';
+import { Header } from './components/Header';
+import { ErrorInfo } from './components/ErrorInfo';
+import { Footer } from './components/Footer';
 
-const USER_ID = 0;
+const getFilteredTodos = (todos: Todo[], filter: FilterType): Todo[] => {
+  const filteredTodos = [...todos];
+
+  switch (filter) {
+    case FilterType.active:
+      return filteredTodos.filter(todo => !todo.completed);
+    case FilterType.completed:
+      return filteredTodos.filter(todo => todo.completed);
+    default:
+      return filteredTodos;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setLoading] = useState(true);
+  const [error, setError] = useState('');
+  const [title, setTitle] = useState('');
+  const [isFocusing, setIsFocusing] = useState(true);
+  const [filter, setFilter] = useState(FilterType.all);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadTodos, setLoadTodos] = useState<number[]>([]);
+
+  const filteredTodos = getFilteredTodos(todos, filter);
+
+  const getCountActive = (): number => {
+    return todos.filter(todo => !todo.completed).length;
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(response => {
+        setTodos(response);
+        setLoading(false);
+      })
+      .catch(() => setError('Unable to load todos'));
```

Comment:
If the request is unsuccessful, the loader state will not change.
```suggestion
    setLoading(true)
    getTodos()
      .then(setTodos)
      .catch(() => setError('Unable to load todos'))
      .finally(() => setLoading(false));
```